distributed/scheduler.py

```diff
@@ -6697,7 +6697,7 @@ async def get_story(self, keys=()):
 
     transition_story = story
 
-    def reschedule(self, key=None, worker=None):
+    def reschedule(self, key=None, worker=None, stimulus_id=None):
```

Comment:
Not counting tests, I can see that this method is invoked only
- from the worker, through RPC
- directly, from stealing.py

In both cases a stimulus_id is available. So you could change this to mandatory.